File: app/Models/VendorApplication.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class VendorApplication extends Model
{
    protected $fillable = [
        'user_id',
        'business_name',
        'business_type',
        'description',
        'status',
        'rejection_reason',
        // Financial Stability Data
        'annual_revenue',
        'net_worth',
        'years_in_business',
        'credit_rating',
        'has_bankruptcy',
        'has_tax_liens',
        // Reputation Data
        'average_rating',
        'total_reviews',
        'has_legal_issues',
        'legal_issues_description',
        'has_complaints',
        'complaints_description',
        // Regulatory Compliance Data
        'has_business_license',
        'has_tax_registration',
        'has_insurance',
        'insurance_type',
        'has_environmental_compliance',
        'has_safety_certification',
        // Facility Visit Data
        'scheduled_visit_date',
        'visit_status',
        'visit_notes',
        'visit_passed',
        // PDF Document Data
        'pdf_document_path',
        'submitted_at',
        'validated_at',
        // Validation Scores
        'financial_score',
        'reputation_score',
        'compliance_score',
        'overall_score',
        'reviewed_by',
        'reviewed_at'
    ];

    protected $casts = [
        'annual_revenue' => 'decimal:2',
        'net_worth' => 'decimal:2',
        'years_in_business' => 'integer',
        'has_bankruptcy' => 'boolean',
        'has_tax_liens' => 'boolean',
        'average_rating' => 'decimal:1',
        'total_reviews' => 'integer',
        'has_legal_issues' => 'boolean',
        'has_complaints' => 'boolean',
        'has_business_license' => 'boolean',
        'has_tax_registration' => 'boolean',
        'has_insurance' => 'boolean',
        'has_environmental_compliance' => 'boolean',
        'has_safety_certification' => 'boolean',
        'scheduled_visit_date' => 'datetime',
        'visit_passed' => 'boolean',
        'submitted_at' => 'datetime',
        'validated_at' => 'datetime',
        'reviewed_at' => 'datetime',
        'financial_score' => 'decimal:2',
        'reputation_score' => 'decimal:2',
        'compliance_score' => 'decimal:2',
        'overall_score' => 'decimal:2'
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function reviewer()
    {
        return $this->belongsTo(User::class, 'reviewed_by');
    }

    // Scopes for filtering
    public function scopePending($query)
    {
        return $query->where('status', 'pending');
    }

    public function scopeApprovedForVisit($query)
    {
        return $query->where('status', 'approved_for_visit');
    }

    public function scopeConditionalApproval($query)
    {
        return $query->where('status', 'conditional_approval');
    }

    public function scopeRejected($query)
    {
        return $query->where('status', 'rejected');
    }

    public function scopeVisitScheduled($query)
    {
        return $query->where('visit_status', 'scheduled');
    }

    public function scopeVisitCompleted($query)
    {
        return $query->where('visit_status', 'completed');
    }

    // Helper methods
    public function isApprovedForVisit()
    {
        return $this->status === 'approved_for_visit';
    }

    public function isConditionallyApproved()
    {
        return $this->status === 'conditional_approval';
    }

    public function isRejected()
    {
        return $this->status === 'rejected';
    }

    public function hasVisitScheduled()
    {
        return $this->scheduled_visit_date !== null;
    }

    public function getStatusBadgeClass()
    {
        switch ($this->status) {
            case 'approved_for_visit':
                return 'badge-success';
            case 'conditional_approval':
                return 'badge-warning';
            case 'rejected':
                return 'badge-danger';
            case 'pending':
            default:
                return 'badge-info';
        }
    }

    public function getVisitStatusBadgeClass()
    {
        switch ($this->visit_status) {
            case 'completed':
                return 'badge-success';
            case 'scheduled':
                return 'badge-warning';
            case 'pending':
                return 'badge-info';
            default:
                return 'badge-secondary';
        }
    }
} 

File: vendor-validation-server/src/main/java/com/scms/model/VendorApplication.java
package com.scms.model;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "vendor_applications")
public class VendorApplication {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String businessName;
    private String businessType;
    private String description;
    private String status;
    private String rejectionReason;
    
    // Financial Stability Data
    private Double annualRevenue;
    private Double netWorth;
    private Integer yearsInBusiness;
    private String creditRating;
    private Boolean hasBankruptcy;
    private Boolean hasTaxLiens;
    
    // Reputation Data
    private Double averageRating;
    private Integer totalReviews;
    private Boolean hasLegalIssues;
    private String legalIssuesDescription;
    private Boolean hasComplaints;
    private String complaintsDescription;
    
    // Regulatory Compliance Data
    private Boolean hasBusinessLicense;
    private Boolean hasTaxRegistration;
    private Boolean hasInsurance;
    private String insuranceType;
    private Boolean hasEnvironmentalCompliance;
    private Boolean hasSafetyCertification;
    
    // Facility Visit Data
    private LocalDateTime scheduledVisitDate;
    private String visitStatus;
    private String visitNotes;
    private Boolean visitPassed;
    
    // PDF Document Data
    private String pdfDocumentPath;
    private LocalDateTime submittedAt;
    private LocalDateTime validatedAt;
    
    // Validation Scores
    private Double financialScore;
    private Double reputationScore;
    private Double complianceScore;
    private Double overallScore;
    
    // Constructors
    public VendorApplication() {}
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public String getBusinessName() { return businessName; }
    public void setBusinessName(String businessName) { this.businessName = businessName; }
    
    public String getBusinessType() { return businessType; }
    public void setBusinessType(String businessType) { this.businessType = businessType; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    
    public String getRejectionReason() { return rejectionReason; }
    public void setRejectionReason(String rejectionReason) { this.rejectionReason = rejectionReason; }
    
    public Double getAnnualRevenue() { return annualRevenue; }
    public void setAnnualRevenue(Double annualRevenue) { this.annualRevenue = annualRevenue; }
    
    public Double getNetWorth() { return netWorth; }
    public void setNetWorth(Double netWorth) { this.netWorth = netWorth; }
    
    public Integer getYearsInBusiness() { return yearsInBusiness; }
    public void setYearsInBusiness(Integer yearsInBusiness) { this.yearsInBusiness = yearsInBusiness; }
    
    public String getCreditRating() { return creditRating; }
    public void setCreditRating(String creditRating) { this.creditRating = creditRating; }
    
    public Boolean getHasBankruptcy() { return hasBankruptcy; }
    public void setHasBankruptcy(Boolean hasBankruptcy) { this.hasBankruptcy = hasBankruptcy; }
    
    public Boolean getHasTaxLiens() { return hasTaxLiens; }
    public void setHasTaxLiens(Boolean hasTaxLiens) { this.hasTaxLiens = hasTaxLiens; }
    
    public Double getAverageRating() { return averageRating; }
    public void setAverageRating(Double averageRating) { this.averageRating = averageRating; }
    
    public Integer getTotalReviews() { return totalReviews; }
    public void setTotalReviews(Integer totalReviews) { this.totalReviews = totalReviews; }
    
    public Boolean getHasLegalIssues() { return hasLegalIssues; }
    public void setHasLegalIssues(Boolean hasLegalIssues) { this.hasLegalIssues = hasLegalIssues; }
    
    public String getLegalIssuesDescription() { return legalIssuesDescription; }
    public void setLegalIssuesDescription(String legalIssuesDescription) { this.legalIssuesDescription = legalIssuesDescription; }
    
    public Boolean getHasComplaints() { return hasComplaints; }
    public void setHasComplaints(Boolean hasComplaints) { this.hasComplaints = hasComplaints; }
    
    public String getComplaintsDescription() { return complaintsDescription; }
    public void setComplaintsDescription(String complaintsDescription) { this.complaintsDescription = complaintsDescription; }
    
    public Boolean getHasBusinessLicense() { return hasBusinessLicense; }
    public void setHasBusinessLicense(Boolean hasBusinessLicense) { this.hasBusinessLicense = hasBusinessLicense; }
    
    public Boolean getHasTaxRegistration() { return hasTaxRegistration; }
    public void setHasTaxRegistration(Boolean hasTaxRegistration) { this.hasTaxRegistration = hasTaxRegistration; }
    
    public Boolean getHasInsurance() { return hasInsurance; }
    public void setHasInsurance(Boolean hasInsurance) { this.hasInsurance = hasInsurance; }
    
    public String getInsuranceType() { return insuranceType; }
    public void setInsuranceType(String insuranceType) { this.insuranceType = insuranceType; }
    
    public Boolean getHasEnvironmentalCompliance() { return hasEnvironmentalCompliance; }
    public void setHasEnvironmentalCompliance(Boolean hasEnvironmentalCompliance) { this.hasEnvironmentalCompliance = hasEnvironmentalCompliance; }
    
    public Boolean getHasSafetyCertification() { return hasSafetyCertification; }
    public void setHasSafetyCertification(Boolean hasSafetyCertification) { this.hasSafetyCertification = hasSafetyCertification; }
    
    public LocalDateTime getScheduledVisitDate() { return scheduledVisitDate; }
    public void setScheduledVisitDate(LocalDateTime scheduledVisitDate) { this.scheduledVisitDate = scheduledVisitDate; }
    
    public String getVisitStatus() { return visitStatus; }
    public void setVisitStatus(String visitStatus) { this.visitStatus = visitStatus; }
    
    public String getVisitNotes() { return visitNotes; }
    public void setVisitNotes(String visitNotes) { this.visitNotes = visitNotes; }
    
    public Boolean getVisitPassed() { return visitPassed; }
    public void setVisitPassed(Boolean visitPassed) { this.visitPassed = visitPassed; }
    
    public String getPdfDocumentPath() { return pdfDocumentPath; }
    public void setPdfDocumentPath(String pdfDocumentPath) { this.pdfDocumentPath = pdfDocumentPath; }
    
    public LocalDateTime getSubmittedAt() { return submittedAt; }
    public void setSubmittedAt(LocalDateTime submittedAt) { this.submittedAt = submittedAt; }
    
    public LocalDateTime getValidatedAt() { return validatedAt; }
    public void setValidatedAt(LocalDateTime validatedAt) { this.validatedAt = validatedAt; }
    
    public Double getFinancialScore() { return financialScore; }
    public void setFinancialScore(Double financialScore) { this.financialScore = financialScore; }
    
    public Double getReputationScore() { return reputationScore; }
    public void setReputationScore(Double reputationScore) { this.reputationScore = reputationScore; }
    
    public Double getComplianceScore() { return complianceScore; }
    public void setComplianceScore(Double complianceScore) { this.complianceScore = complianceScore; }
    
    public Double getOverallScore() { return overallScore; }
    public void setOverallScore(Double overallScore) { this.overallScore = overallScore; }
} 